setup.py

```diff
@@ -91,20 +91,24 @@
     'decorator',
     'requests',
 ]
-
 # Extra dependencies
 EXTRAS_REQUIRES = {
     'tests': [
         'packaging',
         'pyproj',
         'pytest',
-        'pytest-cov',
         'pytest-json-report',
     ],
-    # arclink decryption also works with: pycrypto, m2crypto, pycryptodome
-    'arclink': ['cryptography'],
+    'geo': ['geographiclib'],
+    'imaging': ['cartopy'],
     'io.shapefile': ['pyshp'],
 }
+EXTRAS_REQUIRES['all'] = [dep for depl in EXTRAS_REQUIRES.values()
+                          for dep in depl]
+# arclink decryption also works with: pycrypto, m2crypto, pycryptodome
+# We do not include it in 'all', because the module is deprecated.
+EXTRAS_REQUIRES['arclink'] = ['cryptography']
```

Comment:
If arclink is causing any trouble or work, could also just delete it for good, it's been deprecated for years and my guess is no servers are even up to talk to, main servers have been shut down years ago. If people *really* need it, they can always just install an old obspy version to use it.